### PR TITLE
Feature/support new flips

### PR DESCRIPTION
**Pull request by ToruiDev**

### CODE REVIEW

**Comment by ToruiDev** [comment]
This needs a review and testing online.
Maybe get like 2-3 People to test it out before sharing and merging it.

File: src/main/java/de/torui/coflsky/FlipHandler.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+				t.purge();
+			}
+			if (!Flips.isEmpty()) {
+				CurrentTask = new TimerTask() {
+					@Override
+					public void run() {
+						RunHouseKeeping();
+					}
+				};
+				t.schedule(CurrentTask, 50 * 1000/* 30 seconds */ + /* small arbitrary delay */150);
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/FlipHandler.java:61]: move the seconds amount into a variable

**Review by Ekwav — CHANGES_REQUESTED** [review]

File: src/main/java/de/torui/coflsky/WSCommandHandler.java [hunk 2]
```diff
@@ -41,13 +44,26 @@ public static boolean HandleCommand(JsonStringCommand cmd, Entity sender) {
...
 
 		return true;
 	}
 
+	private static void Flip(Command<FlipData> cmd) {
+		//handle chat message
+		ChatMessageData[] messages = cmd.getData().Messages;
+		Command<ChatMessageData[]> showCmd = new Command<ChatMessageData[]>(CommandType.ChatMessage, messages);
+		ChatMessage(showCmd);
+		flipHandler.fds.Insert(new de.torui.coflsky.FlipHandler.Flip(cmd.getData().Id, cmd.getData().Worth));
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/WSCommandHandler.java:61]: When holding the key down there is a noticeable delay between receiving flips (they even show up in chat) and the auction opening.
I am pretty sure that it is longer than my ping.
Could the opening be triggered in this method to optimise for opening auctions faster as it is very important

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Review by Ekwav — CHANGES_REQUESTED** [review]
Please send the command `track ("bestHotkey {uuid}")` when the hotkey event triggers

File: src/main/java/de/torui/coflsky/network/WSClient.java [hunk 3]
```diff
@@ -122,15 +122,16 @@ public void onTextMessage(WebSocket websocket, String text) throws Exception{
 	}
 
 	public void SendCommand(Command cmd) {
-		Send(cmd);
+		SendCommand(new RawCommand(cmd.getType().ToJson(),gson.toJson(cmd.getData())));
 	}
-
+
```
Comment (Ekwav) [inline src/main/java/de/torui/coflsky/network/WSClient.java:127]:
```suggestion

```

**Review by Ekwav — CHANGES_REQUESTED** [review]

**Review by Ekwav — APPROVED** [review]
LGTM! :)

**Pull request merged**